controller_manager/test/hwi_switch_test.cpp

```diff
@@ -151,19 +152,24 @@ class SwitchBot : public hardware_interface::RobotHW
 
         for(std::list<hardware_interface::ControllerInfo>::const_iterator it = start_list.begin(); it != start_list.end(); ++it)
         {
-            for (std::set<std::string>::const_iterator res_it = it->resources.begin(); res_it != it->resources.end(); ++res_it)
+            if (it->claimed_resources.size() != 1)
+            {
+                ROS_FATAL("We expect controllers to claim resoures from only one interface. This should never happen!");
```

Comment:
This test only exercises the joint mode switching behavior, which is orthogonal to controllers with resources from multiple interfaces. For simplicity, I think it's fine to preserve the assumption that this test works only with controllers whose resources come from a single interface.

I did the minimal changes to this test to make it compatible with this PR.

Still, to take your feedback into account, I'll add a comment in the test stating its assumptions, to prevent potential confusion.